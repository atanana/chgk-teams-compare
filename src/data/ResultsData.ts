export class Tour {
    tour: number;
    answers: number[];

    constructor(tour: number, answers: number[]) {
        this.tour = tour;
        this.answers = answers;
    }
}

export class TeamData {
    id: number;
    name: string;
    city: string;
    tours: Tour[];

    constructor(id: number, name: string, city: string) {
        this.id = id;
        this.name = name;
        this.city = city;
        this.tours = [];
    }
}