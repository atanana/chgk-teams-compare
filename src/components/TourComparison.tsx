import * as React from 'react';
import { TeamData } from '../data/ResultsData';
import { teamText } from '../utils/UiUtils';

interface Props {
    teams: TeamData[];
}

const TourComparison = ({ teams }: Props) => (
    <table className="table is-fullwidth is-hoverable is-striped">
        <thead>
        <th/>
        {
            teams.map(team => (
                <th>{teamText(team)}</th>
            ))
        }
        </thead>
    </table>
);

export default TourComparison;