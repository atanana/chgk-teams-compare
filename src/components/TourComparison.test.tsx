import * as React from 'react';
import TourComparison from './TourComparison';
import * as renderer from 'react-test-renderer';
import { TeamData, Tour, TourComplexity } from '../data/ResultsData';

it('renders without crashing', () => {
    const team1 = new TeamData(1, 'test 1', 'test 1');
    team1.tours = [
        new Tour(1, [1, 0, 1, 0, 0, 1, 1]),
        new Tour(2, [1, 0, 1, 0, 0, 1, 1]),
        new Tour(3, [1, 0, 1, 0, 0, 1, 1])
    ];
    const team2 = new TeamData(2, 'test 2', 'test 2');
    team2.tours = [
        new Tour(1, [0, 0, 1, 1, 0, 0, 1]),
        new Tour(2, [0, 0, 1, 1, 0, 0, 1]),
        new Tour(3, [0, 0, 1, 1, 0, 0, 1])
    ];
    const answerComplexity = [
        new TourComplexity(1, [1, 0.22, 0, 0.33, 0.5, 0.65, 0.34]),
        new TourComplexity(2, [1, 0.22, 0, 0.33, 0.5, 0.65, 0.34]),
        new TourComplexity(3, [1, 0.22, 0, 0.33, 0.5, 0.65, 0.34]),
    ];
    const component = renderer.create(<TourComparison teams={[team1, team2]} answerComplexity={answerComplexity}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});