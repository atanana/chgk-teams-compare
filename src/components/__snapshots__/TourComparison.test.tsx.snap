// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<table
  className="table is-fullwidth is-hoverable is-striped"
>
  <thead>
    <tr>
      <th />
      <th>
        <abbr
          title="100% - (количество взявших вопрос команд)"
        >
          Сложность
        </abbr>
      </th>
      <th>
        <abbr
          title="test 1(test 1)"
        >
          test 1
        </abbr>
      </th>
      <th>
        <abbr
          title="test 2(test 2)"
        >
          test 2
        </abbr>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="has-text-centered"
        colSpan={100}
      >
        Тур 
        1
      </td>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        100.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        22.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        0.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        4
      </td>
      <td>
        33.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        5
      </td>
      <td>
        50.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        6
      </td>
      <td>
        65.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        7
      </td>
      <td>
        34.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td
        className="has-text-centered"
        colSpan={100}
      >
        Тур 
        2
      </td>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        100.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        22.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        0.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        4
      </td>
      <td>
        33.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        5
      </td>
      <td>
        50.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        6
      </td>
      <td>
        65.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        7
      </td>
      <td>
        34.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td
        className="has-text-centered"
        colSpan={100}
      >
        Тур 
        3
      </td>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        100.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        22.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        0.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        4
      </td>
      <td>
        33.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        5
      </td>
      <td>
        50.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        6
      </td>
      <td>
        65.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-danger"
        >
          <i
            className="fa fa-times-circle-o"
          />
        </span>
      </td>
    </tr>
    <tr>
      <td>
        7
      </td>
      <td>
        34.00%
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
      <td>
        <span
          className="tag is-rounded is-success"
        >
          <i
            className="fa fa-check-circle-o"
          />
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
